pkg/openstackbaremetalset/baremetalhost.go

```diff
@@ -217,7 +217,7 @@ func BaremetalHostProvision(
 		if foundBaremetalHost.Status.Provisioning.State != metal3v1.StateProvisioned {
 			foundBaremetalHost.Spec.Image = &metal3v1.Image{
 				URL:          localImageURL,
-				Checksum:     fmt.Sprintf("%s.sha256", localImageURL),
+				Checksum:     fmt.Sprintf("%s.%s", localImageURL, *instance.Spec.OSImageChecksumSuffix),
 				ChecksumType: metal3v1.SHA256,
```

Comment:
Looks good.

Should we use `metal3v1.AutoChecksum` for `ChecksumType` ?

Reply:
It would be nice if we could at least eventually use `metal3v1.AutoChecksum`.  We were intending to deal with that concern though in a follow-up and just get this PR merged to unblock things for what we have for now, which is just sha256.